### DISCUSSION TITLE
Is the data connector extension in llama-index supported?

### DISCUSSION THREAD

**Discussion by xiaolibuzai-ovo** [safevideo/autollm#166] [Q&A] [answered]
there are a lot of data connector extension for llama-index, and is autollm support them? such as from database or other
<img width="1321" alt="image" src="https://github.com/safevideo/autollm/assets/96616876/c0210945-0639-4967-a130-679251d516e1">

**Comment by talhasarit** [comment, marked as answer]
Hi @xiaolibuzai-ovo,

Yes, If you prefer to use the readers implemented in llama-hub and not in our builtin readers, then you can indeed use those as well. For example for database reader, you would use the example given:

```python
from llama_index import download_loader

DatabaseReader = download_loader('DatabaseReader')

reader = DatabaseReader(
    scheme = "postgresql", # Database Scheme
    host = "localhost", # Database Host
    port = "5432", # Database Port
    user = "postgres", # Database User
    password = "FakeExamplePassword", # Database Password
    dbname = "postgres", # Database Name
)

query = f"""
SELECT
    CONCAT(name, ' is ', age, ' years old.') AS text
FROM public.users
WHERE age >= 18
"""

documents = reader.load_data(query=query)
```

Then you can use our `AutoQueryEngine` to build a pipeline and ask questions over your data:

```python
from autollm AutoQueryEngine

query_engine = AutoQueryEngine.from_defaults(documents)
response = query_engine.query("your query")

print(response.response)
```
